src/sim_engine/engine.rs

```
@@ -71,12 +72,19 @@ impl Engine for SimEngine {
     }
 
     fn destroy_pool(&mut self, name: &str) -> EngineResult<()> {
+        let entry = match self.pools.entry(name.into()) {
+            Entry::Vacant(_) => return Ok(()),
```

Trying to destroy a pool that doesn't exist should return an error.